torchaudio/datasets/musdb_hq.py

```diff
@@ -0,0 +1,104 @@
+import os
+from pathlib import Path
+from typing import Tuple, Union, List, Optional
+
+import torch
+import torchaudio
+from torch.utils.data import Dataset
+
+_URL = "https://zenodo.org/record/3338373/files/musdb18hq.zip"
+_EXT = ".wav"
+_SAMPLE_RATE = 44100
+
+
+class MUSDB_HQ(Dataset):
+    """Create *MUSDB_HQ Dataset* [:footcite:`MUSDB18HQ`] Dataset
+
+    Args:
+        root (str or Path): Root directory where the dataset's top level directory is found
+        subset (str): Subset of the dataset to use. Options: [``"train"``, ``"test"``].
+        sources (List[str] or None, optional): Sources extract data from.
+            Options: [``bass``, ``drums``, ``other``, ``mixture``, ``vocals``].
+            If ``None``, dataset consists of tracks except mixture.
+            (default: ``"[``bass``, ``drums``, ``other``, ``vocals``]"``)
+        split (str, optional): Whether to split training set into train and validation set.
+            If None, no splitting occurs. If ``train`` or ``validation``, returns respective set.
+            (default: ``None``)
+        validation (List[str] or None, optional): Tracks included in the validation set, to be used for splitting.
+            List must be provided if `split` is provided, or `None` if `split` is not provided.
+            (default: ``None``)
+    """
+
+    def __init__(
+        self,
+        root: Union[str, Path],
+        subset: str,
+        sources: List[str] = ["bass", "drums", "other", "vocals"],
+        split: Optional[str] = None,
+        validation: Optional[List[str]] = None,
```

Comment:
I dont think this should be a param, as this is hard to use without detailed knowledge of the dataset. 

One idea: have subset `train` `valid`, `full_train` or `test` and if it is `valid` or `train`, use a hardcoded list of the tracks that I sent. This would make the API mucher simpler.

If a someone still wants a custom split, they can find the indexes in `self.names` and build  a torch data `Subset` with those, but my take is that 99% of the users will use the standard splits.

Reply:
is there some reference that shows that the default valid set [here](https://github.com/sigsep/sigsep-mus-db/blob/master/musdb/configs/mus.yaml#L44) is the standard used for musdb-hq? since the paper only splits it into train/test, having `train` refer to a subset of the train can come up to confusing as me, which is why we created a further split param (which is also what is done in the repo linked above). I do see how it can be annoying to enter the validation list so maybe we could use a better default?

@faroit any thoughts on this?

Reply:
The repo linked is the official musdb repository (see https://sigsep.github.io/datasets/musdb.html#tools). However I agree with your point that renaming the train set might be confusing. Then maybe the best is just to follow the musdb API with the extra parameter. The list of validation tracks given in the link should be automatically used though.